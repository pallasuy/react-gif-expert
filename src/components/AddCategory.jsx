import { useState } from 'react';
import { PropTypes } from 'prop-types';

export const AddCategory = ({ onNewCategorie }) => {

  const [inputValue, setInputValue] = useState('');

  const onInputChange = ({ target: { value } }) => {
    setInputValue(value);
  }


  const onSubmit = (event) => {

    event.preventDefault();
    const inputValueT = inputValue.trim();
    if (inputValueT.length < 1) return;//Si la cadena esta vacia no continuo

    // setCategories(categories => [inputValue , ...categories]);//Recibo en el callback las categorias actuales y copio el arreglo luego de la neuva categoria para no perder los datos.
    onNewCategorie(inputValueT);
    setInputValue('');
  }

  return (
    <form onSubmit={event => onSubmit(event)} aria-label='form'>
      <input
        type="text"
        placeholder="Buscar Gifs"
        value={inputValue}
        onChange={(event) => onInputChange(event)}
      />
    </form>
    //Si tenes un callback que recibe un argumento y se lo pasa a una funcion podemos resumirlo como : 

    /*  <form onSubmit={onSubmit}>
     <input
       type="text"
       placeholder="Buscar Gifs"
       value={inputValue}
       onChange={(event) => onInputChange(event)}
     />
   </form> */

  );
}

AddCategory.propTypes = {
  onNewCategorie: PropTypes.func.isRequired,
}

