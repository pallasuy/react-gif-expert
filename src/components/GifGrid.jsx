import { useEffect, useState } from "react";

import { GifItem } from "./GifItem";
import { getGifs } from "../helpers/getGifs";
import { useFetchGifs } from "../hooks/useFetchGifs";
import { PropTypes } from "prop-types";

export const GifGrid = ({ category }) => {

    /*UseEffect recibe 2 argumentos primero el effecto que es un callback y luego lasendencias, si se deja vacio el segundo
    parametro, se ejecutara solo en el constructor de nuestro componente */

    /*  const [images, setImages] = useState([]);
 
     const getImages = async () => {
         const newImages = await getGifs(category);
         setImages(newImages);
     }
 
     useEffect(() => {
         getImages();
     }, []);
  */

    const { isLoading, images } = useFetchGifs(category);



    return (
        <>
            <h3> {category} </h3>

            {
                isLoading && <h2>Cargando...</h2>

            }

            <div className="card-grid">
                {
                    images.map((image) => (
                        <GifItem
                            key={image.id}
                            {...image} //Exparcir las porpiedades es decir envio todas al componente

                        />
                    ))
                }

            </div>
        </>
    )
}

GifGrid.propTypes = {
    category: PropTypes.string.isRequired,

}
