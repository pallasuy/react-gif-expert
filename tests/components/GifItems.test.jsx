/* import {render,screen} from "@testing-library/react";
import { GifItem } from "../src/components/GifItem";

render(<GifItem/>);
screen.debug(); */

import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/components/GifItem";

describe('Prueba en <GifItem/>', () => {

    const title = 'Esto es un titulo';
    const url = 'http://www.google.com/';

    test('Debe hacer match con el snapshot', () => {

        const { container } = render(<GifItem title={title} url={url} />);
        expect(container).toMatchSnapshot();

    });


    test('Debe mostrar la imagen con el URL y el ALT indicado', () => {

        render(<GifItem title={title} url={url} />) //Primero agarramos nuestro objeto de prueba
        //screen.debug(); //Esto para visaulizar nuestro objeto de prueba.
        expect(screen.getByRole('img').src).toBe(url);//Evaluamos propiedad de un elemento. 
        expect(screen.getByRole('img').alt).toBe(title);

    });

    test('Debe mostrar el titulo en el componente', () => {
        render(<GifItem title={title} url={url} />) //Primero agarramos nuestro objeto de prueba
        expect(screen.getByText(title)).toBeTruthy();

    })

});