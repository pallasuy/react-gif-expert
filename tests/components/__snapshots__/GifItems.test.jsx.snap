// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba en <GifItem/> Debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Esto es un titulo"
      src="http://www.google.com/"
    />
    <p>
      Esto es un titulo
    </p>
  </div>
</div>
`;
